src/FilenameParsing/FileIDHelper.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace SilverStripe\Assets\FilenameParsing;
+
+/**
+ * Helps build and parse FileIDs according to a predefined format.
+ */
+interface FileIDHelper
+{
+
+    /**
+     * Map file tuple (hash, name, variant) to a filename to be used by flysystem
+     *
+     * @param string|ParsedFileID $filename Name of file or ParsedFileID object
+     * @param string $hash Hash of original file
+     * @param string $variant (if given)
+     * @return string Adapter specific identifier for this file/version
+     */
+    public function buildFileID($filename, $hash = null, $variant = null);
+
+
+    /**
+     * Performs filename cleanup before sending it back.
```

Comment:
Maybe this should be more clear on what sort of cleanup is to be made to the filename?

Reply:
I've added a somewhat more prescriptive description.

> Clean up filename to remove constructs that might clash with the underlying path format of this FileIDHelper.